### DISCUSSION TITLE
Audit MongoDb 22 no working

### DISCUSSION THREAD

**Discussion by vteixeira87** [thepirat000/Audit.NET#640] [Q&A]
I'm trying to use Audit.NET.MongoDB, but it's not generating the log.

I've already tried it in the same DB as my application, in a separate DB called Audit and Collection Event, but regardless of the configuration it doesn't work.

I tested these 2 examples

Audit.Core.Configuration.DataProvider = new Audit.MongoDB.Providers.MongoDataProvider()
{
    ConnectionString = "mongodb://localhost:27017",
    Database = "Audit",
    Collection = "Event"
};

and

Audit.Core.Configuration.Setup()
    .UseMongoDB(config => config
        .ConnectionString("mongodb://localhost:27017")
        .Database("Audit")
        .Collection("Event"));

My item insert works, but the log is not generated.

I'm using .net 6 and version 22 of the package and I added the examples to the application startup.

Would you help me?

**Comment by thepirat000** [comment]
What are you trying to audit? How are you generating the audit events?

What happens if you manually create and save an audit event after your MongoDB configuration? Do you get an exception?

For exanple:

```c#
Audit.Core.Configuration.Setup()
    .UseMongoDB(config => config
    .ConnectionString("mongodb://localhost:27017")
    .Database("Audit")
    .Collection("Event"));

AuditScope.Log("Event Type", new { ExtraField = "extra value" });
```

**Reply by vteixeira87** [reply]
Many thanks for the reply.

I'm trying to capture all inserts, changes and deletes in the DB. I used the lib with EF Core and only registered it at startup and captured all interactions, I don't know if it should be different in MongoDB.

When I save the event manually there is no error, it is saved successfully in the database.

**Reply by thepirat000** [reply]
I still don't understand what are you trying to audit.

Note `Audit.Core.Configuration.Setup()...` only set up the audit output. 

i.e. `Audit.NET.MongoDb` lib is to save audits to MongoDB, not to audit a MongoDB database. 

All packages starting with "Audit.NET.*" are output extensions.
With the exception of `Audit.EntityFramework` which can do both things: generate audits for a dbcontext and store the audits in a dbcontext.

If you're using Audit.EntityFramework.Core to audit your Dbcontext please share your audit configuration.

**Reply by vteixeira87** [reply]
I'm trying to save a log of the operations I perform in MongoDB, insert, update and delete. 

At startup I only did the example configuration. Do I have to add anything else to save the record? If so, what would it be, would you have an example? 

Testing inserting an event manually did not give any errors. 

Thank you in advance for making the lib available.

**Reply by thepirat000** [reply]
How do you perform the operations in Mongo DB? I'm assuming via MongoDB.Driver.

There is no extension to automatically log interaction with MongoDB.Driver within Audit.NET.


Here you can see all the interaction extensions:
https://github.com/thepirat000/Audit.NET?tab=readme-ov-file#extensions

**Reply by vteixeira87** [reply]
Got it, thank you very much for the answers. 

Congratulations on the project.

**Comment by thepirat000** [comment]
I'll keep this open as I believe we have the opportunity to develop a MongoDB.Driver interaction extension using [command events](https://mongodb.github.io/mongo-csharp-driver/2.8/reference/driver_core/events/). 

I will delve into it further and potentially initiate work on the new extension

**Comment by thepirat000** [comment]
#641

**Comment by thepirat000** [comment]
Nota that a new library is provided [Audit.MongoClient](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.MongoClient/README.md) to support auditing a MongoClient instance.
